codechain_primitives/HexString.py

```diff
@@ -0,0 +1,137 @@
+from rlp import encode
+import string
+
+
+class _HexString():
+    @classmethod
+    def checkString(cls, value):
+        if type(value) is not str:
+            raise ValueError("Value is not string")
+
+        checkValue = value[2:] if value.startswith("0x") else value
+        if not len(checkValue) == cls.length * 2:
+            raise ValueError(
+                f"Expected {cls.length} byte hexstring for creating H{cls.length * 8} but found {value}")
+
+        if all(c in string.hexdigits for c in checkValue):
+            return True
+
+        return False
+
+    @classmethod
+    def fromBytes(cls, buffer):
+        data = buffer[1:]
+        first = buffer[0] if type(buffer[0]) is int else ord(buffer[0])
+
+        if cls is H512:
+            length = data[0] if type(data[0]) is int else ord(data[0])
+            data = data[1:]
+            if first != 0xb8 or length != 64 or len(data) != length:
+                raise ValueError(f"Invalid RLP for H512: {buffer}")
+        elif cls is H128:
+            length = first - 0x80
+            if length != 16 or len(data) != length:
+                raise ValueError(f"Invalid RLP for 128: {buffer}")
+        elif cls is H160:
+            length = first - 0x80
+            if length != 20 or len(data) != length:
+                raise ValueError(f"Invalid RLP for 160: {buffer}")
+        elif cls is H256:
+            length = first - 0x80
+            if length != 32 or len(data) != length:
+                raise ValueError(f"Invalid RLP for 256: {buffer}")
+        else:
+            raise ValueError(f"Invalid RLP for HexString: {buffer}")
+
+        return cls(''.join(hex(c if type(c) is int else ord(c))[2:].zfill(2) for c in data))

```

Comment:
I think an internal representation of the hex should be size limited buffer. It should subclass `Buffer` like `UnsignedInteger` subclasses `int`.
1. It is more compact in memory.
2. It cannot be misrepresented unless the size is wrong.
3. Interop with the code outside will be much easier since many libraries and codes can accept `Byte` type object.

Reply:
I implemented it with bytearray which is immutable type of bytes.